crates/swc_ecma_parser/src/lexer/mod.rs

```diff
@@ -132,6 +132,8 @@ pub struct Lexer<'a> {
     module_errors: Rc<RefCell<Vec<Error>>>,
 
     buf: Rc<RefCell<String>>,
+
+    atoms: Rc<RefCell<AtomStore>>,
```

Comment:
It seems this always does `self.atoms.borrow_mut().atom(...)`. Is this a critical path? Maybe consider creating an `AtomStoreCell` that wraps an UnsafeCell and has a safe API? It should be faster because it will skip the borrow check.

```rs
pub struct AtomStoreCell(UnsafeCell<AtomStore>);

impl AtomStoreCell {
  pub fn atom(&self, ...etc...) {
    // ...etc...
  }
}
```

Reply:
I'm not sure if worths `unsafe`.

Reply:
I've noticed a decent performance improvement when doing it in scenarios where it's borrowed a lot. The important part is to keep the interface safe.

Reply:
If so, let's do it with another PR

Reply:
I just tested it out and in this case the performance was the same.

Reply:
Actually, there is a slight improvement on the "three" benchmark. I will open a PR.